inst/unitTests/cpp/sparse.cpp

```diff
@@ -77,3 +77,41 @@ Rcpp::List sparseList(Rcpp::List l) {
     
     return Rcpp::List::create(mat1, mat2);
 }
+
+// [[Rcpp::export]]
+arma::sp_mat dtc2dgc(SEXP S){
+    S4 mat = S;
+    IntegerVector dims = mat.slot("Dim");
+    int nrow = dims[0];
+    int ncol = dims[1];
+    
+    // Creating an empty SpMat
```

Comment:
I think what we want to test is `arma::sp_mat res = Rcpp::as<arma::sp_mat>(S);`

The converting code has been added into the `as` implementation and we want to test.